datadog/tests/resource_datadog_app_builder_app_test.go

```diff
@@ -0,0 +1,1035 @@
+		default = "Created using the Datadog provider in Terraform. Variable interpolation."
+	}
+	variable "root_instance_name" {
+		type    = string
+		default = "grid0"
+	}
+	variable "connection_id" {
+		type    = string
+		default = "11111111-2222-3333-4444-555555555555"
+	}
+	
+	resource "datadog_app_builder_app" "test_app_escaped_interpolated" {
+		app_json = jsonencode(
+			%s
+		)
+	}`, name, testAppBuilderAppEscapedInterpolatedAppJSON())
+}
+
+func testAppBuilderAppEscapedInterpolatedAppJSON() string {
+	return `
+    {
+		"queries" : [
+			{
+				"id" : "11111111-1111-1111-1111-111111111111",
+				"name" : "listTeams0",
+				"type" : "action",
+				"properties" : {
+					"onlyTriggerManually" : false,
+					"outputs" : "$${((outputs) => {// Use 'outputs' to reference the query's unformatted output.\n\n// TODO: Apply transformations to the raw query output\n\nreturn outputs.data.map(item => item.attributes.name);})(self.rawOutputs)}",
+					"spec" : {
+						"fqn" : "com.datadoghq.dd.teams.listTeams",
+						"inputs" : {},
+						"connectionId" : "${var.connection_id}"
+					}
+				}
+			}
+		],
+		"id" : "11111111-2222-3333-4444-555555555555",
+		"components" : [
+			{
+				"events" : [],
+				"name" : "grid0",
+				"properties" : {
+					"children" : []
+				},
+				"type" : "grid"
+			}
+		],
+		"description" : "${var.description}",
+		"favorite" : false,
+		"name" : "${var.name}",
+		"rootInstanceName" : "${var.root_instance_name}",
+		"selfService" : false,
+		"tags" : [],
+		"connections" : [
+			{
+				"id" : "${var.connection_id}",
+				"name" : "A connection that will be overridden"
+			}
+		],
+		"deployment" : {
+			"id" : "11111111-1111-1111-1111-111111111111",
+			"app_version_id" : "00000000-0000-0000-0000-000000000000"
+		}
+    }
+	`
+}
+
+func testLoadFromFileAppBuilderAppResourceConfig(fileName string) string {
+	return fmt.Sprintf(`
+	resource "datadog_app_builder_app" "test_app_from_file" {
+		app_json = file("%s")
+		override_action_query_names_to_connection_ids = {
+			"listTeams0" = "11111111-2222-3333-4444-555555555555"
+		}
+	}`, fileName)
+}
+
+func testAccCheckDatadogAppBuilderAppExists(accProvider *fwprovider.FrameworkProvider, n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		apiInstances := accProvider.DatadogApiInstances
+		auth := accProvider.Auth
+
+		if err := datadogAppBuilderAppExistsHelper(auth, s, apiInstances, n); err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func datadogAppBuilderAppExistsHelper(ctx context.Context, s *terraform.State, apiInstances *utils.ApiInstances, name string) error {
+	idString := s.RootModule().Resources[name].Primary.ID
+	id, err := uuid.Parse(idString)
+	if err != nil {
+		return fmt.Errorf("error parsing ID as UUID: %s", err)
+	}
+	if _, _, err := apiInstances.GetAppBuilderApiV2().GetApp(ctx, id); err != nil {
+		return fmt.Errorf("error getting app: %s", err)
+	}
+	return nil
+}
+
+func testComplexAppBuilderApp(name string) string {
```

Comment:
can you pull this from the example resource JSON file? i think that's the same as this and it would de-duplicate like 600 lines from this PR.